src/backend/joanie/core/utils/contract.py

```diff
@@ -34,7 +36,17 @@ def _get_base_signature_backend_references(
         **extra_filters,
     ).select_related("order")
 
-    if course_product_relation:
```

Comment:
I think this change is useless. All should work as expected with the current implementation.

Reply:
Indeed.